aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -791,10 +791,10 @@ void cpu_flash_attention(
       // Move to the next query
       data_index_step(i, batchSize, j, num_head, k, qSlice);
     }
+    if (need_pack) {
+      cpublas::brgemm_release();
```

Comment:
How exactly does this work? Now you're repeatedly calling this release inside a parallel_for. Is that really what you want?

Reply:
Yes, release_hw_context need to be called at the end of the parallel section to guarentee that: use AMX within parallel region, and that the state of the CPU is restored afterwards.

Reply:
> How exactly does this work? Now you're repeatedly calling this release inside a parallel_for. Is that really what you want?

Yes - the state if per core.